src/mscorlib/src/System/Random.cs

```diff
@@ -115,11 +122,39 @@ public Random()
           return retVal;
       }
 
+
+      [ThreadStatic]
+      private static Random t_threadRandom;
+      private static Random s_globalRandom = new Random(GenerateGlobalSeed());
+
+      /*=====================================GenerateSeed=====================================
+      **Returns: An integer that can be used as seed values for consecutively
+                 creating lots of instances on the same thread within a short period of time.
+      ========================================================================================*/
+      private static int GenerateSeed() {
+          if (t_threadRandom == null) {
+              int seed;
+              lock (s_globalRandom) {
+                  seed = s_globalRandom.Next();
+              }
+              t_threadRandom = new Random(seed);
+          }
+          return t_threadRandom.Next();
```

Comment:
Unless things have changed recently around thread statics, it'd be good to cache t_threadRandom into a local so that the common path of it already being initialized has only one TLS read, rather than one for the null check and another for the Next call.